src/compressed_tensors/quantization/lifecycle/helpers.py

```diff
@@ -51,3 +54,28 @@ def enable_quantization(module: Module):
 
 def disable_quantization(module: Module):
     module.quantization_enabled = False
+
+
+def safe_permute(value: torch.Tensor, perm: torch.Tensor, dim: int = 0) -> torch.Tensor:
+    """
+    Perform out-of-place permutation without using torch.Tensor.index_put_,
+    whose implementation is missing for datatypes such as `torch.float8_e4m3fn`
+
+    :param value: tensor to permute
+    :param perm: permutation map
+    :param dim: dimension along which to apply permutation
+    :return: permuted value
+    """
+    preceding_dims = [slice(None)] * dim
+    if value.dtype not in EXPERIMENTAL_DTYPES:
+        return value[tuple(preceding_dims + [perm])]
```

Comment:
Rather then keeping a list of experimental types, it might be simpler to simply catch and exception and use the fallback method in that case. Since we are planning on expanding float point quantization support this list is going to get longer, and I think having to update it leaves room for error

Reply:
is there a way to check `index_put_` attr exists in a given value.d_type?

Reply:
I think I'll cache the results of the `try except`, that way we only need to check once for each dtype that is called